SYCL/ESIMD/api/functional/ctors/ctor_default.cpp

```diff
@@ -84,52 +82,53 @@ struct const_ref {
   }
 };
 
-// Functor with the main test routine to iterate over the pre-determined
-// datatypes
-template <typename DataT, int NumElems, typename TestCaseT> struct test {
-  bool operator()(sycl::queue &queue, const std::string &data_type) {
-    bool passed{true};
-
+// Struct that calls simd in provided context and then verifies obtained result.
+struct Verifier {
+  template <typename DataT, int NumElems, typename TestCaseT>
+  static bool verify(sycl::queue &queue, const std::vector<DataT> ref_data,
+                     const std::string &data_type) {
+    bool passed = true;
     DataT default_val{};
-    using AllocatorT = sycl::usm_allocator<DataT, sycl::usm::alloc::shared>;
 
-    std::vector<DataT, AllocatorT> result{NumElems, AllocatorT{queue}};
+    shared_vector<DataT> result(NumElems, shared_allocator<DataT>{queue});
```

Comment:
```suggestion
    shared_vector<DataT> result(NumElems, shared_allocator<DataT>(queue));
```
Here and elsewhere

Reply:
I've applied this comment in this commit: e87ad2ec4b8cff0070719ae8992d50d244c2a019